test/app-tap/datetime.test.lua

```diff
@@ -1830,6 +1830,565 @@ test:test("Parse strptime format", function(test)
     end
 end)
 
+-- Parse string with a custom format, then format datetime object
+-- to a string with the same format and compare initila and
+-- resulted strings:
+--
+--   dt = datetime.parse(<str>, { format = <fmt> })
+--   dt:format(<fmt>) == <str>
+--
+test:test("Parse strptime format and check with :format()", function(test)
+    local formats = {
+        -- %a
+        -- Works for 'Thu' only when date is not complete.
+        {
+            buf = 'Thu',
+            fmt = '%a',
+        },
+        {
+            buf = '19-08-2023 Sat',
+            fmt = '%d-%m-%Y %a',
+        },
+        {
+            buf = '20-08-2023 Sun',
+            fmt = '%d-%m-%Y %a',
+        },
+        {
+            buf = '21-08-2023 Mon',
+            fmt = '%d-%m-%Y %a',
+        },
+        {
+            buf = '22-08-2023 Tue',
+            fmt = '%d-%m-%Y %a',
+        },
+        {
+            buf = '23-08-2023 Wed',
+            fmt = '%d-%m-%Y %a',
+        },
+        {
+            buf = '24-08-2023 Thu',
+            fmt = '%d-%m-%Y %a',
+        },
+        {
+            buf = '25-08-2023 Fri',
+            fmt = '%d-%m-%Y %a',
+        },
+        {
+            buf = '26-08-2023 Sat',
```

Comment:
What is the difference between this test case and the second one (`19-08-2023`)?
Why do we need them both?
Here and for similar tests below.

Minor: Why don't start from Monday | Sunday? :)